File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {
    private static int CARS_PER_HOUR = 221;

    public double productionRatePerHour(int speed) {
        double successRate = 1.0;

        if (speed == 10) {
            successRate = 0.77;
        } else if (speed == 9) {
            successRate = 0.8;
        } else if (speed >= 5) {
            successRate = 0.9;
        }

        return speed * CARS_PER_HOUR * successRate;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) (productionRatePerHour(speed) / 60);
    }
}

File: solutions/fsharp/cars-assemble/1/CarsAssemble.fs
module CarsAssemble

let successRate (speed: int) : float =
    match speed with
    | 10 -> 0.77
    | s when s >= 9 -> 0.80
    | s when s >= 5 -> 0.90
    | s when s >= 1 -> 1.00
    | _ -> 0.0


let productionRatePerHour (speed: int) : float = float (221 * speed) * successRate speed

let workingItemsPerMinute (speed: int) : int = int (productionRatePerHour speed) / 60
